ddtrace/llmobs/_experiment.py

```diff
@@ -256,7 +259,14 @@ def _process_record(self, idx_record: Tuple[int, DatasetRecord]) -> Optional[Tas
                 span_id, trace_id = "", ""
             input_data = record["input_data"]
             record_id = record.get("record_id", "")
-            tags = {"dataset_id": self._dataset._id, "dataset_record_id": record_id, "experiment_id": self._id}
+            tags = deepcopy(self._tags)
```

Comment:
we don't need to `deepcopy()` here, which is a lot more expensive, since the dict doesn't have nested fields. You can simply do `tags = {**self._tags}`

Reply:
TIL that strings are fine to shallow copy 💀